### ISSUE TITLE
Setting PIPENV_DEFAULT_THREE=1 causes env to be recreated every time

### ISSUE THREAD

**Issue by simonpercivall** [kennethreitz/pipenv#412]
It seems like the environment variables for defaulting pipenv to a python version interferes with its logic for _recreating_ a virtualenv.

### How to reproduce (for instance, not necessarily minimal):
```shell
> PIPENV_DEFAULT_THREE=1 pipenv shell
Creating a Pipfile for this project...
Creating a virtualenv for this project...
…
> PIPENV_DEFAULT_THREE=1 pipenv shell
Virtualenv already exists!
Removing existing virtualenv...
Creating a virtualenv for this project...
…
```

**Comment by nateprewitt** [comment]
Thanks for bringing this up @simonpercivall. It definitely looks like we recreate the environment every time when this variable is set. Previously `three` was only set to True when `--three` was passed, meaning we only created the environment once. Now it's passed every time and because it's checked in `ensure_project` every command will create a new virtualenv.

@kennethreitz, could you give a quick rundown of what this feature is adding? I'm not sure I understand what #394 was supposed to change.

**Comment by kennethreitz** [comment]
Discussed on IRC. 

tl;dr: this feature is buggy, and will be removed, for now, until revisited.

**Comment by nateprewitt** [comment]
5.1.0 has been released. We'll revisit this feature in a future release. Thanks again @simonpercivall!
